microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIConfigFileHealthTest.java

```diff
@@ -0,0 +1,206 @@
+            FailSafeDummyService.LIVE_CONFIG_PROPERTY, false,
+            FailSafeDummyService.READY_CONFIG_PROPERTY, true,
+            FailSafeDummyService.IN_MAINTENANCE_CONFIG_PROPERTY, true,
+            FailSafeDummyService.READY_IN_MAINTENANCE_CONFIG_PROPERTY, false);
+
+    protected RequestSpecification metricsRequest;
+
+    @ArquillianResource
+    protected ManagementClient managementClient;
+
+    @Before
+    public void before() throws ConfigurationException, InterruptedException, TimeoutException, IOException {
+        ArquillianContainerProperties arqProps = new ArquillianContainerProperties(
+                ArquillianDescriptorWrapper.getArquillianDescriptor());
+        String url = "http://" + arqProps.getDefaultManagementAddress() + ":" + arqProps.getDefaultManagementPort()
+                + "/metrics";
+        metricsRequest = given().baseUri(url).accept(ContentType.JSON);
+        // reset MP Config properties
+        new Administration(ManagementClientProvider.onlineStandalone(managementClient)).reload();
+    }
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        return ShrinkWrap
+                .create(WebArchive.class, FailSafeCDIHealthBaseTest.class.getSimpleName() + ".war")
+                .addClasses(FailSafeDummyService.class, CDIBasedLivenessHealthCheck.class, CDIBasedReadinessHealthCheck.class)
+                .addAsManifestResource(new StringAsset(MPConfigContent), "microprofile-config.properties")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    /**
+     * @tpTestDetails Multiple-component customer scenario - health check status is based on fail-safe CDI bean and MP Config
+     *                property. There is IOException since service is in maintenance but fallback method should be used.
+     * @tpPassCrit Overall and the health check status are down and fallback method is invoked. MP Metrics are increased
+     *             according to the specification.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Ignore("WFLY-12924, WFLY-12925")
+    @Test
+    final public void testHealthEndpointDownInMaintenance() throws Exception {
+        // TODO Java 11 Map<String, String> liveCheck = Map.of( "name", "dummyLiveness", "status", "DOWN");
+        Map<String, String> liveCheck = Collections.unmodifiableMap(new HashMap<String, String>() {
+            {
+                put("name", "dummyLiveness");
+                put("status", "DOWN");
+            }
+        });
+        Map<String, String> readyCheck = Collections.unmodifiableMap(new HashMap<String, String>() {
+            {
+                put("name", "dummyReadiness");
+                put("status", "DOWN");
+            }
+        });
+        get(HealthUrlProvider.healthEndpoint()).then()
+                .statusCode(503)
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
```

Comment:
Aren't
```
contentType(ContentType.JSON)
```
and
```
header("Content-Type", containsString("application/json"))
```
a different way to check for the same header value? Also in following test methods' checks. Just checking here.

Reply:
Yeah, it is the same...I guess it has came from Throntail TS, will fix...